### PR TITLE
Refactor frontend and migrate to flux-container@2.1

### PR DESCRIPTION
**Pull request by Ma27**
### Overview
- [x] use flux-container@2.1 and flux-react@2.0
- [x] simplify store structure
- [x] redesign view
  - [x] improve `Menu` structure
  - [x] minor refactorings of form related components
### Related issues

related to #338 
### BC breaks

None.

### CODE REVIEW

**Comment by Ma27** [comment]
@benbieler This is now ready to review (finally :D)

File: travis-ci/install.sh [hunk 1]
```diff
@@ -2,9 +2,9 @@
 
 set -e
 
-# setup for node 6
+# setup for node 7
 sudo rm -rf ~/.nvm
-curl -sL https://deb.nodesource.com/setup_6.x | sudo -E bash -
+curl -sL https://deb.nodesource.com/setup_7.x | sudo -E bash -
```
Comment (benbieler) [inline travis-ci/install.sh]: I dont know. I would stick with the LTS version. (6.9)

Comment (Ma27) [reply travis-ci/install.sh]: yep AFAIK the `setup_6.x` script configures the PPA to stick on the latest LTS.
I'd propose to create an env variable (`NODE_VERSION` for instance) in the `.travis.yml`, so we can check against Node.js 6.x and Node.js 7.x?

Comment (Ma27) [reply travis-ci/install.sh]: I think that we'll do somehow the same when PHP 7.1 will be released.

Comment (benbieler) [reply travis-ci/install.sh]: Ok. Fine

**Review by benbieler — APPROVED** [review]

**Review by Ma27 — COMMENTED** [review]
LGTM

**Pull request merged**